shell/platform/windows/flutter_windows_view.cc

```diff
@@ -49,10 +53,37 @@ void FlutterWindowsView::SetEngine(
                     binding_handler_->GetDpiScale());
 }
 
-void FlutterWindowsView::OnWindowSizeChanged(size_t width,
-                                             size_t height) const {
-  surface_manager_->ResizeSurface(GetRenderTarget(), width, height);
+uint32_t FlutterWindowsView::GetFrameBufferId(size_t width, size_t height) {
+  std::unique_lock<std::mutex> lock(resize_mutex_);
+
+  if (resize_status_ != ResizeState::kResizeStarted) {
+    return kWindowFrameBufferID;
+  }
+
+  if (resize_target_width_ == width && resize_target_height_ == height) {
+    surface_manager_->ResizeSurface(GetRenderTarget(), width, height);
+    surface_manager_->MakeCurrent();
```

Comment:
Do we actually need these two steps to done within the lock? It's unclear to me if the mutex is also intended to protect the actual buffer swap, or just the ivars.

Relatedly, it would be helpful to annotate all these lock-using methods with the thread we expect them to be called on so it's easier to understand the flow.

Reply:
`surface_manager_` methods here don't need to be protected by the mutex, the line immediately after this needs to be protected. That being said having these be protected by the mutex shouldn't cause any contention (for the flow as it stands now) given that all the surface manager methods will be invoked on the raster task runner.


I had a question about validating the task runner, i see this pattern a lot: `GetTaskRunner()->RunsTasksOnCurrentThread()`, but it looks like in the case of windows the emedding only has a handle to the platform task runner / thread. I added comments to the methods with the task runner they run on, but i suspect theres something better we can do. Any ideas?

Reply:
Yes, we only have access to the task runners of threads we provide custom runners for. Comments are fine here; my concern wasn't enforcing invariants, just helping readers understand the threading flow (the assertion version is nice because then you have confidence that the comments aren't out of sync with reality, but I don't think that's a significant concern here.)

Reply:
> That being said having these be protected by the mutex shouldn't cause any contention (for the flow as it stands now) given that all the surface manager methods will be invoked on the raster task runner.

Just to make sure I'm understanding: the reason we don't care about the lock blocking the platform thread unnecessarily during these steps is that if this block is running the platform thread is blocked on this whole step completing regardless, right?

It wouldn't put a comment here to that effect so that it doesn't look like a locking mistake (which tends to be my default concern when I see non-trivial function calls happening inside a lock).

Reply:
Added a comment clarifying that platform thread is blocked for the entirety of the step.